components/script/dom/html/htmlinputelement.rs

```diff
@@ -2380,6 +2385,22 @@ impl HTMLInputElement {
             .SetData(placeholder_text);
     }
 
+    pub(crate) fn select_files_for_webdriver(
+        &self,
+        test_paths: Vec<DOMString>,
+        response_sender: IpcSender<Result<bool, ErrorStatus>>,
+    ) {
+        let mut stored_sender = self.pending_webdriver_response.borrow_mut();
+        assert!(stored_sender.is_none());
+
+        *stored_sender = Some(PendingWebDriverResponse {
+            response_sender,
+            expected_file_count: test_paths.len(),
+        });
```

Comment:
if file picker selection is now asynchronous, does that mean Element Send Keys handling for file inputs is asynchronous too? what happens if a client sends two requests for the same input element?